### PR TITLE
Adding implementation for the action

### PR DESCRIPTION
**Pull request by JosephDSchwartz**
# Summary of PR changes

Creates a new action that will query Jira for a task using filters given in the parameters. It then checks the task to make sure all of the appropriate approvals are given.

## PR Requirements
- [X] The examples in the repository's `README.md` have been updated with the new version.  
  - See the *Incrementing the Version* section of the repository's README.md for more details on how the version will be incremented.
- [X] The action code does not contain sensitive information.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,80 +1,51 @@
...
-  - [ ] Update the Outputs section with the correct action outputs
-  - [ ] Update the Example section with the correct usage   
-- action.yml
-  - [ ] Fill in the correct name, description, inputs and outputs and implement steps
-- CODEOWNERS
-  - [ ] Update as appropriate
+## TODOs 
 - Repository Settings
-  - [ ] On the *Options* tab check the box to *Automatically delete head branches*
   - [ ] On the *Options* tab update the repository's visibility
```
Comment (danielle-casella-adams) [inline README.md]: These can be removed now along with the entry in the Index

File: README.md [hunk 2]
```diff
@@ -1,80 +1,51 @@
...
-| `input`   | true        | Description goes here |
-
-## Outputs
-| Output   | Description           |
-| -------- | --------------------- |
-| `output` | Description goes here |
+| Parameter                          | Is Required | Default | Description                                                                                                                                                          |
+| ---------------------------------- | ----------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `domain-name`                      | true        | N/A     | The domain name for Jira.                                                                                                                                            |
+| `proposed-build-unique-identifier` | true        | N/A     | A string that will be compared against the proposed build field in order to identify the deployment ticket. This can be the whole value or just a unique part of it. |
```
Comment (danielle-casella-adams) [inline README.md]: Will this make sense to the person using this action? It doesn't make sense to me. :)

File: README.md [hunk 3]
```diff
@@ -1,80 +1,51 @@
...
+| `projects-to-filter-by`            | false       | N/A     | A comma separated string of project names to filter tickets by.                                                                                                      |
+| `skip-qa-approval`                 | false       | false   | A flag determining whether or not to skip checking for QA approval. Valid values are "true" and "false".                                                             |
+| `check-subtasks`                   | false       | false   | A flag determining whether or not to check for deployment sub-tasks. Valid values are "true" and "false".                                                            |
```
Comment (danielle-casella-adams) [inline README.md]: I wonder if tweaking the text or the name here might be helpful, I didn't really understand what this was for until I got to the implementation.   If you named it the same thing as our script `deploymentTaskCanBeSubtask` that would clarify what this was for.

File: src/jira-ticket-check.ps1 [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# If multiple issues are found, select the first one
+if ($DeploymentTask.Count -gt 1) {
+    $DeploymentTask = $DeploymentTask[0]
+}
+
+$DeploymentTaskKey = $DeploymentTask.key
+Write-Output "Found the matching Deployment Task $DeploymentTaskKey"
+
+# Technical Approval
+$TechnicalApprovalField = $DeploymentTask.fields.customfield_16323
```
Comment (danielle-casella-adams) [inline src/jira-ticket-check.ps1]: Would this customfield work for all of our Jira projects or just the funding ones?

Comment (JosephDSchwartz) [reply src/jira-ticket-check.ps1]: I'm 99% sure the answer is yes. I queried all of our custom fields and that was the only one that looked related to technical approval. There is a 1% chance there is another custom field with a weird name that I missed, but if my refactor goes well it should handle that scenario.

File: README.md (hunk 2 above)
Comment (JosephDSchwartz) [reply README.md]: It probably would have made sense to the two teams who will use this action, but not to anyone else. I've renamed it to (hopefully) be more helpful.

File: README.md (hunk 3 above)
Comment (JosephDSchwartz) [reply README.md]: I ended up calling it `check-parent-task`. I'm now trying to lean into the idea that you're searching for a task to check in this action. With that in mind, hopefully this property makes more sense.

File: src/jira-ticket-check.ps1 (hunk 4 above)
Comment (JosephDSchwartz) [reply src/jira-ticket-check.ps1]: We're now checking whatever fields are passed in, so if the 1% chance happens then we're still able to use this action!

**Review by hpractv — DISMISSED** [review]
👍🏽

**Review by hpractv — DISMISSED** [review]
👍🏽

File: action.yml [hunk 5]
```diff
@@ -1,22 +1,45 @@
-name: 'Composite Run Steps Action Template'
+name: 'Check Jira Deployment Attestations'
```
Comment (danielle-casella-adams) [inline action.yml]: Should this be renamed?

File: action.yml [hunk 6]
```diff
@@ -1,22 +1,45 @@
...
+    required: false
+    default: 'false'
 
 runs:
   using: 'composite'
   steps:
-    - name: Step 1
-      id: step_1
-      shell: bash
+    - name: Check for attestations on Jira task
```
Comment (danielle-casella-adams) [inline action.yml]: Should this be renamed?

File: action.yml [hunk 7]
```diff
@@ -1,22 +1,45 @@
-name: 'Composite Run Steps Action Template'
+name: 'Check Jira Deployment Attestations'
 
-description: 'Add description here'
+description: 'This action will check a deployment task for technical, qa, and stakeholder approvals. If one is not found then the action will fail.'
```
Comment (danielle-casella-adams) [inline action.yml]: I think the description should be updated.

Comment (JosephDSchwartz) [reply action.yml]: Definitely! And those other things you mentioned as well!

**Review by danielle-casella-adams — APPROVED** [review]

**Pull request merged**